pkg/aggregator/k8sportforward.go

```diff
@@ -0,0 +1,113 @@
+package aggregator
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/iver-wharf/wharf-cmd/pkg/workerapi/workerclient"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+type portForwardedWorker struct {
+	workerclient.Client
+	portConn portConnection
+	podName  string
+}
+
+type portConnection struct {
+	portforward.ForwardedPort
+	stopCh chan struct{}
+}
+
+func newPortForwardedWorker(a k8sAggr, podName string, buildID uint) (portForwardedWorker, error) {
+	portConn, err := newPortForwarding(a, a.namespace, podName)
+	if err != nil {
+		return portForwardedWorker{}, err
+	}
+	worker, err := a.newWorkerClient(portConn.Local, buildID)
+	if err != nil {
+		portConn.Close()
+		return portForwardedWorker{}, err
+	}
+	pfWorker := portForwardedWorker{
+		Client:   worker,
+		portConn: portConn,
+		podName:  podName,
+	}
+	return pfWorker, nil
+}
+
+func (w portForwardedWorker) CloseAll() {
+	w.Client.Close()
+	w.portConn.Close()
+}
```

Comment:
Right, I think the rename was because it didn't conform when it didn't return an error, but I'll just make it return nil.

Not entirely sure if the errors should be ignored though, so will need to reevaluate anyways 👍